applications/MappingApplication/custom_mappers/nearest_element_mapper.h

```diff
@@ -22,42 +22,139 @@
 
 // Project includes
 #include "mapper.h"
+#include "custom_utilities/mapping_matrix_builder.h"
+#include "custom_searching/interface_communicator.h"
+#include "custom_utilities/interface_preprocessor.h"
+#include "custom_utilities/mapper_flags.h"
+#include "custom_utilities/mapper_local_system.h"
 
 
 namespace Kratos
 {
-///@addtogroup ApplicationNameApplication
+///@name Kratos Classes
 ///@{
 
-///@name Kratos Globals
-///@{
+class NearestElementInterfaceInfo : public MapperInterfaceInfo
+{
+public:
 
-///@}
-///@name Type Definitions
-///@{
+    /// Default constructor.
+    NearestElementInterfaceInfo() {}
 
-///@}
-///@name  Enum's
-///@{
+    NearestElementInterfaceInfo(const CoordinatesArrayType& rCoordinates,
+                                const IndexType SourceLocalSystemIndex,
+                                const IndexType SourceRank)
+        : MapperInterfaceInfo(rCoordinates, SourceLocalSystemIndex, SourceRank) {}
 
-///@}
-///@name  Functions
-///@{
+    MapperInterfaceInfo::Pointer Create() const override
+    {
+        return Kratos::make_shared<NearestElementInterfaceInfo>();
+    }
 
-///@}
-///@name Kratos Classes
-///@{
+    MapperInterfaceInfo::Pointer Create(const CoordinatesArrayType& rCoordinates,
+                                        const IndexType SourceLocalSystemIndex,
+                                        const IndexType SourceRank=0) const override
```

Comment:
Avoid default parameters in virtual functions since they get mapped statically.

```suggestion
                                        const IndexType SourceRank) const override
```

If you still want default parameters consider using non-virtual interface for the function:
```
public:
    MapperInterfaceInfo::Pointer Create(const CoordinatesArrayType& rCoordinates,
                                        const IndexType SourceLocalSystemIndex,
                                        const IndexType SourceRank=0) const
   {
        this->CreateImpl(rCoordinates, SourceLocalSystemIndex, SourceRank);
   }
private:
    virtual  MapperInterfaceInfo::Pointer CreateImpl(const CoordinatesArrayType& rCoordinates,
                                        const IndexType SourceLocalSystemIndex,
                                        const IndexType SourceRank) const;
```

Or only put it in the base class.

Reply:
done